cmd/gimlet/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+        parse(try_from_str = parse_int::parse)
+    )]
+    timeout: u64,
+
+    /// UDP port to contact on the target.
+    #[clap(long, default_value_t = HARDCODED_PORT)]
+    port: u16,
+
+    /// IPv6 address, e.g. `fe80::0c1d:9aff:fe64:b8c2%en0`
+    #[clap(
+        long,
+        env = "HUMILITY_RPC_IP",
+        group = "target",
+        use_value_delimiter = true
+    )]
+    ip: String,
+
+    #[clap(subcommand)]
+    cmd: SubCmd,
+}
+
+#[derive(Parser, Debug)]
+enum SubCmd {
+    /// Read the current contents of the sequencer FPGA registers.
+    ReadSeqRegs,
+}
+
+pub struct Client {
+    socket: UdpSocket,
+    // Allocating this overly-large buffer because we're on the big computer
+    // where RAM is cheap, and it's less likely to need adjustment this way.
+    buf: [u8; 1024],
+}
+
+impl Client {
+    pub fn new(ip: &str, port: u16, timeout: Duration) -> Result<Self> {
+        let Some((ip, scope)) = ip.rsplit_once('%') else {
+            bail!("Missing scope id in IP (e.g. '%en0')")
+        };
+
+        let scopeid = decode_iface(scope)?;
+
+        let dest = SocketAddrV6::new(
+            ip.parse()?,
+            port,
+            0, // flow info
+            scopeid,
+        );
+
+        let socket = UdpSocket::bind("[::]:0")?;
+        socket.set_read_timeout(Some(timeout))?;
+        socket.connect(dest)?;
+
+        Ok(Self { socket, buf: [0; 1024] })
+    }
+
+    pub fn read_sequencer_registers(&mut self) -> Result<Vec<u8>> {
+        use gimlet_inspector_protocol::{
+            QueryV0, Request, SequencerRegistersResponseV0,
+        };
+
+        let len = hubpack::serialize(
+            &mut self.buf,
+            &Request::V0(QueryV0::SequencerRegisters),
+        )?;
+
+        self.socket.send(&self.buf[..len])?;
+        let n = self.socket.recv(&mut self.buf).context("receiving packet")?;
+        let buf = &self.buf[..n];
+
+        let (response, extra) =
+            hubpack::deserialize::<SequencerRegistersResponseV0>(buf)?;
+        match response {
+            SequencerRegistersResponseV0::Success => Ok(extra.to_vec()),
+            e => {
+                bail!("failed: {e:?}");
+            }
+        }
+    }
+}
+
+fn run(context: &mut ExecutionContext) -> Result<()> {
+    let Subcommand::Other(subargs) = context.cli.cmd.as_ref().unwrap();
+    let subargs = Args::try_parse_from(subargs)?;
+
+    let mut client = Client::new(
+        &subargs.ip,
+        subargs.port,
+        Duration::from_millis(subargs.timeout),
+    )?;
+
+    // We can generalize this when we have more than one command defined.
+    match subargs.cmd {
+        SubCmd::ReadSeqRegs => match client.read_sequencer_registers() {
+            Ok(bytes) => {
+                println!("Received register contents follow:");
+                let mut dumper = Dumper::new();
+                dumper.size = 1;
+                dumper.dump(&bytes, 0);
+            }
+            Err(e) => bail!(e),
+        },
```

Comment:
Style nit: isn't this `match` equivalent to just
```suggestion
        SubCmd::ReadSeqRegs => {
            let bytes = client.read_sequencer_registers()?;
            println!("Received register contents follow:");
            let mut dumper = Dumper::new();
            dumper.size = 1;
            dumper.dump(&bytes, 0);
        },
```

Reply:
Yeah, reckon so. I wrote it the expanded way when I was expecting to do more in the Err branch, but I'm not sure I need to do anything else right now.

Reply:
yup, makes sense to me! not a big deal either way!